src/features/public/Donations/components/ContactDetails.tsx

```diff
@@ -0,0 +1,203 @@
+            />
+            {errors.firstName && (
+              <span className={styles.formErrors}>
+                First Name field is required
+              </span>
+            )}
+          </div>
+
+          <div style={{ width: '20px' }}></div>
+          <div>
+            <MaterialTextFeild
+              inputRef={register({ required: true })}
+              label="Last Name"
+              variant="outlined"
+              name="lastName"
+              onChange={changeContactDetails}
+              defaultValue={contactDetails.lastName}
+            />
+            {errors.lastName && (
+              <span className={styles.formErrors}>
+                Last Name field is required
+              </span>
+            )}
+          </div>
+        </div>
+        <div className={styles.formRow}>
+          <div style={{ width: '100%' }}>
+            <MaterialTextFeild
+              inputRef={register({
+                required: true,
+                pattern: /^([a-zA-Z0-9_\-\.]+)@([a-zA-Z0-9_\-\.]+)\.([a-zA-Z]{2,5})$/i,
+              })}
+              label="Email"
+              variant="outlined"
+              name="email"
+              onChange={changeContactDetails}
+              defaultValue={contactDetails.email}
+            />
+            {errors.email && (
+              <span className={styles.formErrors}>Email is required</span>
+            )}
+          </div>
+        </div>
+        <div className={styles.formRow}>
+          <div style={{ width: '100%' }}>
+            <MaterialTextFeild
+              inputRef={register({ required: true })}
+              label="Address"
+              variant="outlined"
+              name="address"
+              onChange={changeContactDetails}
+              defaultValue={contactDetails.address}
+            />
+            {errors.address && (
+              <span className={styles.formErrors}>Address is required</span>
+            )}
+          </div>
+        </div>
+        <div className={styles.formRow}>
+          <div>
+            <MaterialTextFeild
+              inputRef={register({ required: true })}
+              label="City"
+              variant="outlined"
+              name="city"
+              onChange={changeContactDetails}
+              defaultValue={contactDetails.city}
+            />
+            {errors.city && (
+              <span className={styles.formErrors}>City is required</span>
+            )}
+          </div>
+
+          <div style={{ width: '20px' }}></div>
+          <div>
+            <MaterialTextFeild
+              inputRef={register({ pattern: /^(0|[1-9][0-9]*)$/i })}
+              label="Zip Code"
+              variant="outlined"
+              name="zipCode"
+              onChange={changeContactDetails}
+              defaultValue={contactDetails.zipCode}
+            />
+            {errors.zipCode && (
+              <span className={styles.formErrors}>
+                ZipCode is should only be Numeric
```

Comment:
Not correct; Postalcode in Canada are alpha numeric. It would be great to implement some validator here.